packages/campaignclassic/android/src/main/java/com/adobe/marketing/mobile/reactnative/optimize/RCTAEPCampaignClassic.java

```diff
@@ -0,0 +1,72 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0 Unless required by applicable law or
+ agreed to in writing, software distributed under the License is distributed on
+ an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing
+ permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile.reactnative.campaignclassic;
+
+import com.adobe.marketing.mobile.AdobeCallback;
+import com.adobe.marketing.mobile.AdobeCallbackWithError;
+import com.adobe.marketing.mobile.AdobeError;
+import com.adobe.marketing.mobile.campaignclassic.CampaignClassic;
+import com.adobe.marketing.mobile.campaignclassic.DecisionScope;
+import com.adobe.marketing.mobile.campaignclassic.Offer;
+import com.adobe.marketing.mobile.campaignclassic.OfferType;
+import com.adobe.marketing.mobile.campaignclassic.Proposition;
+import com.adobe.marketing.mobile.LoggingMode;
+import com.adobe.marketing.mobile.MobileCore;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class RCTAEPCampaignClassicModule extends ReactContextBaseJavaModule {
+
+  private static final String TAG = "RCTAEPCampaignClassicModule";
+  private final ReactApplicationContext reactContext;
+
+  public RCTAEPCampaignClassicModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    this.reactContext = reactContext;
+  }
+
+  @Override
+  public String getName() {
+    return "AEPCampaignClassic";
+  }
+
+  @ReactMethod
+  public void extensionVersion(final Promise promise) {
+    promise.resolve(CampaignClassic.extensionVersion());
+  }
+
+  @ReactMethod
+  public void registerDevice(final String deviceToken, final String userKey,
+                             final Map<String, Object> additionalParams) {
+    CampaignClassic.registerDevice(deviceToken, userKey, additionalParams);
```

Comment:
The Android API has a callback, you might have to pass `null` as a fourth parameter. https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-campaignclassic/adobe-campaignclassic-api-reference#registerdevice

Reply:
Added!